src/Integration.Vsix/CFamily/PreCompiledHeaders/PchFilesDeleter.cs

```diff
@@ -0,0 +1,55 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.IO.Abstractions;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily
+{
+    internal interface IPchFilesDeleter
+    {
+        void DeletePchFiles();
+    }
+
+    internal class PchFilesDeleter : IPchFilesDeleter
+    {
+        private readonly IFileSystem fileSystem;
+        private readonly string pchFilePath;
+
+        public PchFilesDeleter(IFileSystem fileSystem, string pchFilePath)
+        {
+            this.fileSystem = fileSystem;
+            this.pchFilePath = pchFilePath;
+        }
+
+        public void DeletePchFiles()
+        {
+            var pchDirectory = Path.GetDirectoryName(pchFilePath);
+            var pchFileName = Path.GetFileName(pchFilePath);
+            var pchFiles = fileSystem.Directory.GetFiles(pchDirectory, $"{pchFileName}.*");
```

Comment:
Have the CFamily guaranteed that any files they create will always append a "." to our file name, or that just what they do at the moment?
A safer option would be to match "[ourfilename]*" (or even `[ourfilename without extension].*`. 
Our file naming should be unique enough that it won't accidentally pick up any external files.

Reply:
I changed to `[ourfilename]*` as that's our agreement with them so far and it's simpler